### DISCUSSION TITLE
Fetch or stream downloaded torrent content from qBittorrent

### DISCUSSION THREAD

**Discussion by Hamamas** [DrewCarlson/qBittorrent-Kotlin#175] [Ideas]
Is there and way to play the content (mp4,mkv.....) directly with out download the content than play it

**Comment by DrewCarlson** [comment]
Hi @Hamamas, the qBittorrent Web API/UI does not provide endpoints to download/read file content, so there is nothing available in this library to facilitate that.

To implement this, you'd need to deploy a custom server next to the qBittorrent instance that provides an endpoint which can respond with the downloaded file content.  For example on you can call `getTorrentFiles()` for the torrent you're interested in to get the relative path of the file.  Then send a request to your custom server with the file path and (with Ktor server) use `call.respondFile(file)` to provide the content for a supported client (VLC, Video.js, etc).
